charts/px-central/templates/px-backup/pre-upgrade-hook/pre-upgrade-check.yaml

```diff
@@ -1,5 +1,32 @@
 {{- $isOpenshiftCluster := .Capabilities.APIVersions.Has "apps.openshift.io/v1" -}}
 {{- $azureProxyEnabled := .Values.proxy.azureProxyEnabled | default false }}
+
+{{- if .Release.IsUpgrade }}
+  {{- $sts := lookup "apps/v1" "StatefulSet" .Release.Namespace "pxc-backup-mongodb" -}}
+  {{- if $sts }}
+
+    # Expected image from helm chart values
+    {{- $expectedImage := printf "%s/%s/%s:%s" .Values.images.mongodbImage.registry .Values.images.mongodbImage.repo .Values.images.mongodbImage.imageName .Values.images.mongodbImage.tag -}}
+
+    {{- range $container := $sts.spec.template.spec.containers }}
+      {{- if ne $container.image $expectedImage }}
+        # Images do not match, check for ready replicas
+        {{- $readyReplicas := $sts.status.readyReplicas | default 0 -}}
+
+        {{- if lt $readyReplicas 3 }}
+          {{ fail (printf "ERROR MongoDB ready replicas expected = 3, received = %d" $readyReplicas. Please refer to the documentation to bring up all 3 pods.) }}
```

Comment:
Q: should we check with Shilpa and give a link to docs, if at all user checks?
@siva-portworx